src/gp/src/GPMSA.C

```diff
@@ -380,10 +380,11 @@ GPMSAEmulator<V, M>::lnValue(const V & domainVector,
     queso_assert_greater(emulator_data_precision, 0);
     double nugget = 1.0 / emulator_data_precision;
 
-    for (unsigned int disc = 0; disc != num_discrepancy_bases;
-         ++disc)
-      covMatrix(disc*m_numExperiments+i,
-                disc*m_numExperiments+i) += nugget;
+    unsigned int discrepancy_basis_offset =
+      numOutputs == 1 ? 0 : num_discrepancy_bases;
+
+    covMatrix(discrepancy_basis_offset*m_numExperiments+i,
+              discrepancy_basis_offset*m_numExperiments+i) += nugget;
   }
```

Comment:
Wait, this can't be right, can it?  This is supposed to be adding to a diagonal, or at least a diagonal of an on-diagonal block, not a *single* entry?  I thought you had just discussed changing the offset, not getting rid of the for loop.

Reply:
Summarizing offline discussion: it looks like we want to *replace* the for loop; we should be looping over num_svd_terms here, not num_discrepancy_bases?

Reply:
Right.  I was about to summarise our in-person discussion but you nailed it.

Reply:
Done.